src/components/collateDailyRequests/collateRequests.ts

```diff
@@ -0,0 +1,42 @@
+import { ResidentRequestCollation } from "../../types/resident-request-collation";
+
+export async function fetchDailyRequests(): Promise<ResidentRequestCollation[]> {
+  try {
+    const response = await fetch("/api/request-schedule");
+    if (!response.ok) {
+      throw new Error("Failed to fetch requests");
+    }
+    return response.json();
+  } catch (error) {
+    console.error("Error fetching daily requests:", error);
+    throw error;
+  }
+}
+
+export async function collateDailyRequests(): Promise<ResidentRequestCollation[]> {
+  const requests = await fetchDailyRequests();
+
+  const groupedRequests: Record<string, ResidentRequestCollation[]> = {
+    Morning: [],
+    Noon: [],
+    Afternoon: [],
```

Comment:
The three labels for the slots (Morning, Noon, Afternoon) are defined with different terms in src/lib/utils/time-slot/time-slot.ts - should we make these labels more consistent?